gpii/node_modules/flowManager/src/CloudBasedFlowManager.js

```diff
@@ -0,0 +1,245 @@
+                },
+                members: {
+                    expressApp: "{kettle.server}.expressApp"
+                },
+                events: {
+                    onContributeMiddleware: "{kettle.server}.events.onContributeMiddleware",
+                    onContributeRouteHandlers: "{kettle.server}.events.onContributeRouteHandlers"
+                }
+            }
+        },
+        deviceReporterDataSource: {
+            type: "fluid.emptySubcomponent"
+        }
+    },
+    urls: {
+        deviceReporter: ""
+    },
+    handlers: {
+        oauth2Settings: {
+            route: "/settings",
+            type: "get"
+        },
+        userLogin: null,
+        userLogout: null,
+        getUserToken: null
+    }
+});
+
+// TODO: This component is in an absurd namespace
+fluid.defaults("kettle.requests.request.handler.oauth2Settings", {
+    gradeNames: ["fluid.littleComponent", "kettle.requests.request.handler.settings", "autoInit"],
+    invokers: {
+        handle: {
+            funcName: "gpii.request.flowManager.onOauth2Settings",
+            args: ["{request}", "{that}", "{gpii.flowManager.cloudBased.oauth2}.oauth2DataStore"],
+            dynamic: true
+        }
+    },
+    listeners: {
+        processMatch: { // Contribute our filter into the middle of the chain
+            priority: gpii.request.flowManager.processMatch.priorities.privacyFilter,
+            namespace: "privacyFilter",
+            funcName: "gpii.flowManager.cloudBased.privacy.filter",
+            args: ["{arguments}.0", "{request}.selectedPreferences"]
+        }
+    }
+});
+
+fluid.registerNamespace("gpii.flowManager.cloudBased.privacy");
+
+// TODO: Restore the ability of the ontology manager to straightforwardly form single transforms and remove this hard-coded path 
+
+gpii.flowManager.cloudBased.privacy.reverseTransform = fluid.require("${universal}/testData/ontologies/privacy-flat.json");
+gpii.flowManager.cloudBased.privacy.forwardTransform = fluid.model.transform.invertConfiguration(gpii.flowManager.cloudBased.privacy.reverseTransform);
+
+gpii.flowManager.cloudBased.privacy.applyFilterRules = function (settings, rules) {
+    var holder = {
+        model: {}
+    };
+    var applier = fluid.makeNewChangeApplier(holder);
+    fluid.each(rules, function (value, path) {
+        if (value) {
+            var fetched = fluid.get(settings, path);
+            applier.change(path, fetched); // TODO: Apparently fluid.set is not good enough to deal with change at root 
+        }
+    });
+    return holder.model;
+};
+
+
+gpii.flowManager.cloudBased.privacy.filterOneApplication = function (application, selectedPreferences) {
+    var settings = application.settings;
+    var settingsInPrivacy = fluid.model.transformWithRules(settings, gpii.flowManager.cloudBased.privacy.forwardTransform);
+    console.log("GOT SETTINGS IN PRIVACY ONTOLOGY " + JSON.stringify(settingsInPrivacy, null, 2));
```

Comment:
We should remove these console logs or replace them with calls to fluid.log before committing to master